pkg/stanza/fileconsumer/design.md

```diff
@@ -87,47 +100,40 @@ Each poll cycle runs through a series of steps which are presented below.
     1. The file system is searched for files with a path that matches the `include` setting.
     2. Files that match the `exclude` setting are discarded.
     3. As a special case, on the first poll cycle, a warning is printed if no files are matched. Execution continues regardless.
-4. Queueing
-    1. If the number of matched files is less than or equal to the maximum degree of concurrency, as defined by the `max_concurrent_files` setting, then no queueing occurs.
-    2. Else, queueing occurs, which means the following:
-        - Matched files are split into two sets, such that the first is small enough to respect `max_concurrent_files`, and the second contains the remaining files (called the queue).
-        - The current poll interval will begin processing the first set of files, just as if they were the only ones found during the matching phase.
-        - Subsequent poll cycles will pull matches off of the queue, until the queue is empty.
-        - The `max_concurrent_files` setting is respected at all times.
-5. Opening
-    1. Each of the matched files is opened. Note:
+4. Opening
+    1. Each of the matched files is opened sequentially. Note:
         - A small amount of time has passed since the file was matched.
         - It is possible that it has been moved or deleted by this point.
         - Only a minimum set of operations should occur between file matching and opening.
         - If an error occurs while opening, it is logged.
-6. Fingerprinting
+5. Fingerprinting
     1. The first `N` bytes of each file are read. (See fingerprinting section above.)
-7. Exclusion
+6. Exclusion
     1. Empty files are closed immediately and discarded. (There is nothing to read.)
     2. Fingerprints found in this batch are cross referenced against each other to detect duplicates. Duplicate files are closed immediately and discarded.
         - In the vast majority of cases, this occurs during file rotation that uses the copy/truncate method. (See fingerprinting section above.)
-8. Reader Creation
+7. Reader Creation
     1. Each file handle is wrapped into a `Reader` along with some metadata. (See Reader section above)
         - During the creation of a `Reader`, the file's fingerprint is cross referenced with previously known fingerprints.
         - If a file's fingerprint matches one that has recently been seen, then metadata is copied over from the previous iteration of the Reader. Most importantly, the offset is accurately maintained in this way.
         - If a file's fingerprint does not match any recently seen files, then its offset is initialized according to the `start_at` setting.
-9. Detection of Lost Files
+8. Detection of Lost Files
```

Comment:
I think behavior described here has been changed in this PR. Moreover, the term "lost files" no longer seems to be an apt description of how we're managing file handles.

What do you think about dropping this terminology and just describing how we hold on to file handles as long as possible, and then read to the end one last time before closing them?

Reply:
Sounds good to me.